File: frontend/src/app/shared/service/collaborator/homework/HomeWorkEtudiantCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {HomeWorkEtudiantDto} from '../../../model/homework/HomeWorkEtudiant.model';
import {HomeWorkEtudiantCriteria} from '../../../criteria/homework/HomeWorkEtudiantCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class HomeWorkEtudiantCollaboratorService extends AbstractService<HomeWorkEtudiantDto, HomeWorkEtudiantCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/homeWorkEtudiant/';
    }

    public constrcutDto(): HomeWorkEtudiantDto {
        return new HomeWorkEtudiantDto();
    }

    public constrcutCriteria(): HomeWorkEtudiantCriteria {
        return new HomeWorkEtudiantCriteria();
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/homework/HomeWorkEtudiantCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.homework;

import java.util.List;
import ma.zs.alc.bean.core.homework.HomeWorkEtudiant;
import ma.zs.alc.dao.criteria.core.homework.HomeWorkEtudiantCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface HomeWorkEtudiantCollaboratorService extends  IService<HomeWorkEtudiant,HomeWorkEtudiantCriteria>  {

    List<HomeWorkEtudiant> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);
    List<HomeWorkEtudiant> findByHomeWorkId(Long id);
    int deleteByHomeWorkId(Long id);
    long countByHomeWorkId(Long id);




}
